File: backend/app/schemas/standard_clause.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class StandardClauseBase(BaseModel):
    standard_id: uuid.UUID
    parent_clause_id: uuid.UUID | None = None
    clause_number: str
    title: str
    summary: str | None = None
    audit_guidance: str | None = None
    evidence_examples: str | None = None
    sort_order: int = 0
    status: str = "active"
    metadata_json: dict | None = None


class StandardClauseCreate(StandardClauseBase):
    tenant_id: uuid.UUID


class StandardClauseUpdate(BaseModel):
    parent_clause_id: uuid.UUID | None = None
    clause_number: str | None = None
    title: str | None = None
    summary: str | None = None
    audit_guidance: str | None = None
    evidence_examples: str | None = None
    sort_order: int | None = None
    status: str | None = None
    metadata_json: dict | None = None


class StandardClauseResponse(StandardClauseBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    created_at: datetime
    updated_at: datetime


class StandardClauseListResponse(BaseModel):
    items: list[StandardClauseResponse]
    total: int

File: frontend/src/types/standardClause.ts
export type StandardClauseRecord = {
  id: string;
  tenant_id: string;
  standard_id: string;
  parent_clause_id: string | null;
  clause_number: string;
  title: string;
  summary: string | null;
  audit_guidance: string | null;
  evidence_examples: string | null;
  sort_order: number;
  status: string;
  metadata_json: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
};

export type StandardClauseListResponse = {
  items: StandardClauseRecord[];
  total: number;
};